### ISSUE TITLE
Error loading preloads: Could not find renderer

### ISSUE THREAD

**Issue by thistlillo** [microsoft/vscode-jupyter#9967]
All of a sudden the jupyter extension has problems with ipywidgets. Yesterday I started getting this error whenever I use `tqdm` and `display(dataframe)` in a notebook. 
```
Error loading preloads:
Could not find renderer
```

Yesterday I had to reinstall the VSC python tools (including vscode-jupyter) and the conda environment I use inside of a remote environment: that is the only change with respect to two days ago, when there were no issues at all. The conda environment was built  using a `yaml` file (`conda env export > env.yaml`) that was used to create the environment in use two days ago.

Both the local and the remote machines run linux.

The VSC extension `Jupyter Notebook Renderers` is installed, version v1.0.6.

I tried to uninstall and reinstall the Jupyter extension (as suggested somewhere), but with no effects.
## Environment data

Version: 1.66.2
Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
Date: 2022-04-11T07:49:24.808Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Linux x64 5.13.0-40-generic

**Comment by rchiodo** [comment]
Thanks for the bug.

What's your version of the jupyter extension?

VS code is also newer than that now. Maybe there's a version conflict.

This is what I'm getting with a simple tqdm example.

![image](https://user-images.githubusercontent.com/19672699/167674475-a3e06132-93c9-490c-9dec-f6b76bd7b0fa.png)

Although I did this locally. Are you connected through say remote ssh?

**Comment by thistlillo** [comment]
Hello and thanks for your anwer.
I was finally able to have it working again, but I needed to uninstall / reinstall / close CSV / reopen VSC.

I think my installation has become a little messy after this round of updates since I have problems with many other extensions. Not even python code autocompletion works anymore.

**Comment by rchiodo** [comment]
Thanks.

**Comment by rchiodo** [comment]
@cpoptic this is not reproing for me. Can you open a new issue? We'd need to look at logs and such to see why the renderers aren't installed.

**Comment by cpoptic** [comment]
I sunk something like 3 hours Uninstalling the Jupyter extension, reloading
VS Code, reinstalling the extension, re running the notebook, getting the
same error, uninstalling the Jupyter notebook Renderers extension,
reloading VS code, reinstalling both Jupter and the Notebook Renderers
extension, rerunning the notebook, etc.

The "Select Kernel" button stops working, and says "Install kernels from
the marketplace" even though I had just run the notebook with a kernel.  So
then you have to uninstall Jupyter extension yet again and reload,
reinstall it....

What a mess.

Eventually thru some random permutation it worked. But I've sunk far too
much time trying to fix this bug and can no longer afford any more.

TLDR, the Jupter extension needs some serious work in stability and
reliability.

**Comment by rchiodo** [comment]
Sorry to hear that. The problem sounds like installs aren't working correctly. Like something isn't being downloaded or the unzip of the VSIX is missing stuff for some reason.

**Comment by thistlillo** [comment]
Exaxtly. The only way to solve the issue was (now I am not experiencing it
any more) to try plenty of permutations of install/unistall/enable/...

Il gio 16 giu 2022, 21:25 Cpop ***@***.***> ha scritto:

> I sunk something like 3 hours Uninstalling the Jupyter extension, reloading
> VS Code, reinstalling the extension, re running the notebook, getting the
> same error, uninstalling the Jupyter notebook Renderers extension,
> reloading VS code, reinstalling both Jupter and the Notebook Renderers
> extension, rerunning the notebook, etc.
>
> The "Select Kernel" button stops working, and says "Install kernels from
> the marketplace" even though I had just run the notebook with a kernel. So
> then you have to uninstall Jupyter extension yet again and reload,
> reinstall it....
>
> What a mess.
>
> Eventually thru some random permutation it worked. But I've sunk far too
> much time trying to fix this bug and can no longer afford any more.
>
> TLDR, the Jupter extension needs some serious work in stability and
> reliability.
>
>

**Comment by rchiodo** [comment]
> how can you get the missing stuff

This is kind of ambiguous, but I'm assuming you're asking how to fix the problem? Since I'm not sure of the root cause of the install not working, we'd need a repro with logs from VS code and more information about your specific situation (proxy in use, where do you download from, etc).

**Comment by GF-Huang** [comment]
Issue still, all jupyter extension is up to date.

![image](https://user-images.githubusercontent.com/4510984/176820209-c12206f2-a151-4bb9-bd3e-507df6508e01.png)

**Comment by GF-Huang** [comment]
> Issue still, all jupyter extension is up to date.
> 
> ![image](https://user-images.githubusercontent.com/4510984/176820209-c12206f2-a151-4bb9-bd3e-507df6508e01.png)


Solved by:

```py
import plotly.io as pio
pio.renderers.default = 'notebook'
```

**Comment by rchiodo** [comment]
> I think the vscode team should consider deleting the extension's old folder when upgrading the extension, otherwise there will be endless issues of this kind.

I believe this happens when you shutdown all instances of VS code. That's when it removes the old extension folders. It can't do that while it's running because the code is in use.

**Comment by GF-Huang** [comment]
> > I think the vscode team should consider deleting the extension's old folder when upgrading the extension, otherwise there will be endless issues of this kind.
> 
> I believe this happens when you shutdown all instances of VS code. That's when it removes the old extension folders. It can't do that while it's running because the code is in use.

Maybe, at least show a tips to user to close all instances before upgrade extensions.

**Comment by QThans** [comment]
**The pre-release version can solve this issue. After I switched to pre-release version, it's ok.**

**Comment by Genozen** [comment]
This issue still exists...
QThans suggested switching to pre-release version which works for me as well...
